src/BrainPass/BrainPass.sol

```diff
@@ -0,0 +1,343 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "openzeppelin-contracts/contracts/token/ERC721/ERC721.sol";
+import {Counters} from "openzeppelin-contracts/contracts/utils/Counters.sol";
+import {Owned} from "solmate/auth/Owned.sol";
+import {SafeMath} from "openzeppelin-contracts/contracts/utils/math/SafeMath.sol";
+
+interface IERC20 {
+    function transferFrom(
+        address from,
+        address to,
+        uint256 amount
+    ) external returns (bool);
+}
+
+/// @title BRAIN Pass NFT
+/// @author Oleanji
+/// @notice A pass for IQ Wiki Editors
+
+contract BrainPassCollectibles is ERC721, Owned {
```

Comment:
Some improvements I've been thinking after seeing the codebase:

- We might want to have some kind of configuration that allow us to change stuff like: min. max minting max time
- We might want to use seconds across the calculations of minting / pricing. That will simplify a lot the code and avoid doing many multiplications


Besides that, there is a missing contract needed that connects the owning of these NFTs with IQ wiki possibility to push wikis

Reply:
Do you mean the Wiki contract part of the ep contracts?

Reply:
the validator / whitelist contracts